CenterForOpenScience/osf.io
# [ENG-2123][ENG-2268] Legacy indicators for storage usage

## Purpose

Let a user know when they are at or approaching a their storage limit.

## Changes

- make sure we are consistently using GB vs. GiB (detour, but since current storage usage is in GBs this really has to be too for ease of formatting)
- pass storage status info into contextvar in the mako template and format
- pass storage status info into fangorn
- throw storage usage error growl on upload 409 response

<img width="621" alt="Screen Shot 2020-09-10 at 2 15 08 PM" src="https://user-images.githubusercontent.com/9688518/93094569-10644c00-f670-11ea-8f30-f03344983629.png">
<img width="515" alt="Screen Shot 2020-09-10 at 2 15 03 PM" src="https://user-images.githubusercontent.com/9688518/93094576-135f3c80-f670-11ea-8a86-2c5b6d6ad6b3.png">
<img width="616" alt="Screen Shot 2020-09-10 at 2 14 13 PM" src="https://user-images.githubusercontent.com/9688518/93094584-15290000-f670-11ea-9e05-643fa566d564.png">
<img width="590" alt="Screen Shot 2020-09-10 at 2 13 53 PM" src="https://user-images.githubusercontent.com/9688518/93094586-15c19680-f670-11ea-8358-e311fa9abe2a.png">

## QA Notes

Note: Since https://openscience.atlassian.net/browse/ENG-2122 is still in progress the growlbox that should appear when a user uploads a file over the limit shouldn't actually be trigger-able. 

  - Does this change require a data migration? If so, what data will we migrate?
No
  - What is the level of risk?
Low
    - Any permissions code touched?
No
    - Is this an additive or subtractive change, other?
additive
  - How can QA verify? (Through UI, API, AdminApp or AdminAdminApp?)
UI
  - What features or workflows might this change impact?
None?
  - How will this impact performance?
It shouldn't noticably

Pages affected:

** Project Page **

<img width="1440" alt="Screen Shot 2020-09-18 at 1 28 34 PM" src="https://user-images.githubusercontent.com/9688518/93627520-10878300-f9b3-11ea-9731-508bbd0fb655.png">

** Files  Page **

<img width="1194" alt="Screen Shot 2020-09-18 at 11 12 50 AM" src="https://user-images.githubusercontent.com/9688518/93627560-1c734500-f9b3-11ea-9c65-e5b67ff09f6a.png">

and check all different places fangorn is used to ensure it's appropriate.

## Documentation

No new docs, I think.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2123
https://openscience.atlassian.net/browse/ENG-2268
